faunadb-java-dsl/src/main/java/com/faunadb/client/types/Encoder.java

```diff
@@ -0,0 +1,205 @@
+package com.faunadb.client.types;
+
+import com.faunadb.client.errors.FaunaException;
+import com.faunadb.client.types.Properties.Property;
+import com.faunadb.client.types.Value.*;
+import com.faunadb.client.types.time.HighPrecisionTime;
+import org.joda.time.Instant;
+import org.joda.time.LocalDate;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Stack;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static com.google.common.collect.Lists.newArrayListWithCapacity;
+import static com.google.common.collect.Maps.newHashMap;
+import static java.lang.String.format;
+
+/**
+ * FaunaDB object to {@link Value} encoder.
+ */
+public class Encoder {
```

Comment:
I think the default encode mechanics should be via `Value` conversion method. Like: 

```java
User user = new User("bob", "pas123");
client.query(Create(Class("users"), Obj("data", Value(user))));
```